sql/core/src/test/resources/sql-tests/inputs/cte.sql

```diff
@@ -155,6 +155,419 @@ SELECT (
+  )
+  SELECT i AS j FROM s
+  UNION ALL
+  SELECT j + 1 FROM t WHERE j < 10
+)
+SELECT * FROM t;
+
+-- recursion nested into recursion 2
+WITH RECURSIVE t AS (
+  WITH RECURSIVE s AS (
+    SELECT j, 1 AS i FROM t
+    UNION ALL
+    SELECT j, i + 1 FROM s WHERE i < 3
+  )
+  VALUES (1) as t(j)
+  UNION ALL
+  SELECT j + 1 FROM s WHERE j < 3
+)
+SELECT * FROM t;
+
+-- routes represented here is as follows:
+--
+-- New York<--->Boston
+-- |            ∧
+-- ∨            |
+-- Washington---+
+-- |
+-- ∨
+-- Raleigh
+CREATE TEMPORARY VIEW routes(origin, destination) AS VALUES
+  ('New York', 'Washington'),
+  ('New York', 'Boston'),
+  ('Boston', 'New York'),
+  ('Washington', 'Boston'),
+  ('Washington', 'Raleigh');
+
+-- handling cycles that could cause infinite recursion
+WITH RECURSIVE destinations_from_new_york AS (
+  SELECT 'New York' AS destination, ARRAY('New York') AS path, 0 AS length
+  UNION ALL
+  SELECT r.destination, CONCAT(d.path, ARRAY(r.destination)), d.length + 1
+  FROM routes AS r
+  JOIN destinations_from_new_york AS d ON d.destination = r.origin AND NOT ARRAY_CONTAINS(d.path, r.destination)
+)
+SELECT * FROM destinations_from_new_york;
+
+-- Fibonacci numbers
+WITH RECURSIVE fibonacci AS (
+  VALUES (0, 1) AS t(a, b)
+  UNION ALL
+  SELECT b, a + b FROM fibonacci WHERE a < 10
+)
+SELECT a FROM fibonacci ORDER BY a;
+
+-- solving Sudoku
+WITH RECURSIVE sudoku AS (
+  VALUES (
+    ARRAY(
+      0, 0, 6, 0, 2, 3, 0, 0, 1,
+      5, 0, 0, 0, 0, 0, 9, 0, 0,
+      0, 0, 2, 0, 0, 4, 0, 0, 0,
+      2, 0, 0, 8, 0, 0, 0, 9, 3,
+      0, 0, 1, 0, 0, 7, 0, 0, 0,
+      8, 3, 0, 0, 0, 0, 4, 0, 0,
+      6, 0, 0, 1, 0, 0, 5, 0, 4,
+      0, 5, 0, 0, 0, 0, 6, 0, 0,
+      0, 0, 9, 0, 7, 0, 0, 2, 0
+    ),
+    0
+  ) AS t(puzzle, level)
+  UNION ALL
+  SELECT
+    CONCAT(SLICE(puzzle, 1, newPosition - 1), ARRAY(newValue), SLICE(puzzle, newPosition + 1, 9 * 9 - newPosition)),
+    level + 1
+  FROM (
+    SELECT
+      puzzle,
+      newPosition,
+      EXPLODE(allowedValues) AS newValue,
+      level
+    FROM (
+      SELECT
+        puzzle,
+        newPosition,
+        ARRAY_EXCEPT(
+          ARRAY_EXCEPT(
+            ARRAY_EXCEPT(
+              SEQUENCE(1, 9),
+              -- used values in row
+              SLICE(puzzle, FLOOR((newPosition - 1) / 9) * 9 + 1, 9)
+            ),
+            -- used values in column
+            TRANSFORM(puzzle, (x, i) -> IF(i % 9 = (newPosition - 1) % 9, x, 0))
+          ),
+          -- used values in 3x3 block
+          TRANSFORM(
+            SLICE(puzzle, FLOOR((newPosition - 1) / (3 * 9)) * 3 * 9 + 1, 3 * 9),
+            (x, i) -> IF(FLOOR(i / 3) % 3 = FLOOR((newPosition - 1) / 3) % 3, x, 0)
+          )
+        ) AS allowedValues,
+        level
+      FROM sudoku
+      JOIN (SELECT EXPLODE(SEQUENCE(1, 9 * 9)) AS newPosition) ON puzzle[newPosition - 1] = 0
+      ORDER BY SIZE(allowedValues)
+      LIMIT 1
+    )
+  )
+)
+SELECT * FROM sudoku WHERE NOT ARRAY_CONTAINS(puzzle, 0);
```

Comment:
IMHO this is a bit complicated for testing.... even if this test fails, we might be not sure about what's the root cause of the failure?

Reply:
Ok, dropped this one.